modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/manager/v1_0/test/DefaultObjectEntryManagerImplTest.java

```diff
@@ -6184,8 +6185,41 @@ public void testMoveObjectEntryToTrash() throws Exception {
 		AssertUtils.assertEquals(
 			WorkflowConstants.STATUS_IN_TRASH, objectEntryVersion.getStatus());
 
+		AssertUtils.assertFailure(
+			ObjectEntryVersionStatusException.CannotExpireEntryInTrash.class,
+			"Cannot expire a version of an entry that is in recycle bin",
+			() -> _defaultObjectEntryManager.expireObjectEntryByVersion(
+				dtoConverterContext, _objectDefinition1, objectEntryId, 1));
+
+		_updateObjectEntryVersion(_objectDefinition1, objectEntry1, 2);
+
+		_defaultObjectEntryManager.deleteObjectEntry(
+			dtoConverterContext, _objectDefinition1, objectEntry1.getId());
+
+		AssertUtils.assertEquals(
+			WorkflowConstants.STATUS_IN_TRASH, objectEntryVersion.getStatus());
+
+		Page<ObjectEntry> objectEntriesPage =
+			_defaultObjectEntryManager.getObjectEntries(
+				_objectDefinition1.getCompanyId(), _objectDefinition1,
+				objectEntry1.getScopeKey(), null, dtoConverterContext, "", null,
+				null, null);
+
+		Collection<ObjectEntry> entries = objectEntriesPage.getItems();
+
+		boolean hasObjectEntryInTrash = entries.stream(
```

Comment:
We cannot use stream in Liferay, please change to another logic, maybe using `ListUtil.filter(`